### PR TITLE
Add cluster profile vsphere-discon

### PR DESCRIPTION
**Pull request by jinyunma**
Add new cluster profile vsphere-discon
related PR in openshift/ci-tools repository is https://github.com/openshift/ci-tools/pull/2253

### CODE REVIEW

**Comment by jinyunma** [comment]
Per doc https://docs.ci.openshift.org/docs/how-tos/adding-a-cluster-profile/#adding-a-new-cluster-profile, need to extract the part of adding new cluster profile from #21245 as a separate PR.

**Comment by jinyunma** [comment]
cc @jcpowermac @petr-muller  for review,  thanks.

File: core-services/ci-secret-bootstrap/_config.yaml [hunk 1]
```diff
@@ -1341,6 +1341,48 @@ secret_configs:
...
+      field: ssh-publickey
+      item: jenkins-ci-iam
+    vmc.secret.auto.tfvars:
+      field: vmc.secret.auto.tfvars
+      item: vsphere-credentials
+  to:
+  - cluster_groups:
+    - non_app_ci
+    name: cluster-secrets-vsphere-discon
+    namespace: ci
```
Comment (petr-muller) [inline core-services/ci-secret-bootstrap/_config.yaml:1385]: So this whole secret has 100% identical content to `cluster-secrets-vsphere`. Is that expected?

Comment (jinyunma) [reply core-services/ci-secret-bootstrap/_config.yaml:1385]: yes, the same content with cluster-secrets-vsphere, the purpose to add new cluster profile(vspehre-discon) is only to use different lease resource. I'm not sure if I can reuse the secrets cluster-secrets-vsphere when configuring job where cluster_profile is vsphere-discon, or must define the section here?

**Comment by jinyunma** [comment]
@stevekuznetsov  @bbguimaraes could you please help to review it ? since Petr is on vacation this week, thanks.

**Comment by bbguimaraes** [comment]
@jinyunma: you configured the profile to have type `vsphere`:
https://github.com/openshift/ci-tools/pull/2253/files#diff-dfee74d02fd350fb1ee1ffbbbc32acf4cb075c9ae42b6e31c8c7acefe7fed623R1182
so it will use the same secret, you don't need to add a new one:
https://github.com/openshift/ci-tools/blob/1b71469e873d2e1a32afa9ede3b68991ec21111d/pkg/prowgen/prowgen.go#L525

You'll see that if you regenerate the jobs in https://github.com/openshift/ci-tools/pull/2253.

**Comment by jinyunma** [comment]
@bbguimaraes thanks for your detailed explanation, understood. I removed that part, please help to review this time.

**Pull request merged**